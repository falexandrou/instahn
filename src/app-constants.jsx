export const BASE_API_URL     = 'https://hacker-news.firebaseio.com/v0';

export const STORIES_SHOW     = 'show';
export const STORIES_BEST     = 'best';
export const STORIES_TOP      = 'top';
export const STORIES_NEW      = 'new';
export const STORIES_ASK      = 'ask';

export const PER_PAGE         = 10;

export const CACHE_NAMESPACE  = 'instahn.offline';