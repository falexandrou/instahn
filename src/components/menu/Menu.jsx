import React from 'react';

import {
  STORIES_TOP,
  STORIES_SHOW,
  STORIES_BEST,
  STORIES_NEW,
  STORIES_ASK,
} from 'app-constants';

import './Menu.scss';

class Menu extends React.Component {
  labels = {
    [STORIES_TOP]: `Top Stories`,
    [STORIES_NEW]: `New on HackerNews`,
    [STORIES_BEST]: `Best Of`,
    [STORIES_SHOW]: `Show HackerNews`,
    [STORIES_ASK]: `Ask HackerNews`,
  };

  render () {
    const { activeType, onSelectType } = this.props;

    let $menuItems = Object.keys(this.labels).map( key => {
      return <li key={`item-${key}`}>
        <a className={ activeType == key ? 'is-active' : '' } onClick={ () => onSelectType(key) }>
          { this.labels[key] }
        </a>
      </li>;
    });

    return <aside className="menu type-selection">
      <ul className="menu-list">{$menuItems}</ul>
    </aside>;
  }
}

export default Menu;