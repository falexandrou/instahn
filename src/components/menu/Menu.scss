@import "../app/variables.scss";

.navigation-menu,
.main-wrapper {
  transition: right 0.2s ease-in-out;
}

.navigation-menu {
  background-color: #fff;
  width: $active-menu-width;
  height: 100%;
  position: fixed;
  top: 0;
  right: -$active-menu-width;
  bottom: 0;
  left: auto;
  z-index: 2;
  border-left: 1px solid $primary;

  .navbar-brand {
    justify-content: center;
  }

  .menu-title {
    justify-content: center;
  }

  .menu, footer {
    margin: 0 2em 2em;
  }

  &.is-open {
    right: 0;

    & + .main-wrapper {
      right: $active-menu-width;
    }
  }

  @media (min-width: $desktop) {
    $offcanvas-width: $active-menu-width / 2;

    width: $offcanvas-width;
    right: -$offcanvas-width;

    &.is-open {
      & + .main-wrapper {
        right: $offcanvas-width;
      }
    }
  }
}