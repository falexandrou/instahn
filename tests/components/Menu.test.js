import 'raf/polyfill';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Menu from 'components/menu/Menu';
import { STORIES_TOP } from 'app-constants';

import { Provider } from 'react-redux';

import { createStore } from '../mocks/store';

const fakeStore = createStore();

describe('Menu Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Provider store={fakeStore}><Menu isOpen={false} activeType={STORIES_TOP} onSelectType={ () => {} } /></Provider>, div);
  });

  it('should match snapshot when the menu is closed', () => {
    const output = renderer.create(<Provider store={fakeStore}><Menu isOpen={false} activeType={STORIES_TOP} onSelectType={ () => {} } /></Provider>).toJSON();
    expect(output).toMatchSnapshot();
  });

  it('should match snapshot when the menu is open', () => {
    const output = renderer.create(<Provider store={fakeStore}><Menu isOpen={true} activeType={STORIES_TOP} onSelectType={ () => {} } /></Provider>).toJSON();
    expect(output).toMatchSnapshot();
  });
});