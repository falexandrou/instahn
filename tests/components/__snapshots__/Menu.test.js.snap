// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component should match snapshot when the menu is closed 1`] = `
<nav
  aria-label="secondary navigation"
  className="navigation-menu "
>
  <div
    className="navbar is-primary"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item hackernews-logo"
        href="/"
      >
        <img
          alt="Hackernews"
          src="/assets/hackernews.svg"
        />
      </a>
    </div>
  </div>
  <div
    className="menu"
  >
    <p
      className="menu-label"
    >
      More HackerNews Stories
    </p>
    <ul
      className="menu-list"
    >
      <li>
        <a
          className="is-active"
          onClick={[Function]}
        >
          Top Stories
        </a>
      </li>
      <li>
        <a
          className=""
          onClick={[Function]}
        >
          New on HackerNews
        </a>
      </li>
      <li>
        <a
          className=""
          onClick={[Function]}
        >
          Best Of
        </a>
      </li>
      <li>
        <a
          className=""
          onClick={[Function]}
        >
          Show HackerNews
        </a>
      </li>
      <li>
        <a
          className=""
          onClick={[Function]}
        >
          Ask HackerNews
        </a>
      </li>
    </ul>
  </div>
  <footer>
    <div
      className="tags has-addons"
    >
      <span
        className="tag"
      >
        Connectivity
      </span>
      <span
        className="tag is-primary"
      >
        Online!
      </span>
    </div>
  </footer>
</nav>
`;

exports[`Menu Component should match snapshot when the menu is open 1`] = `
<nav
  aria-label="secondary navigation"
  className="navigation-menu is-open"
>
  <div
    className="navbar is-primary"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item hackernews-logo"
        href="/"
      >
        <img
          alt="Hackernews"
          src="/assets/hackernews.svg"
        />
      </a>
    </div>
  </div>
  <div
    className="menu"
  >
    <p
      className="menu-label"
    >
      More HackerNews Stories
    </p>
    <ul
      className="menu-list"
    >
      <li>
        <a
          className="is-active"
          onClick={[Function]}
        >
          Top Stories
        </a>
      </li>
      <li>
        <a
          className=""
          onClick={[Function]}
        >
          New on HackerNews
        </a>
      </li>
      <li>
        <a
          className=""
          onClick={[Function]}
        >
          Best Of
        </a>
      </li>
      <li>
        <a
          className=""
          onClick={[Function]}
        >
          Show HackerNews
        </a>
      </li>
      <li>
        <a
          className=""
          onClick={[Function]}
        >
          Ask HackerNews
        </a>
      </li>
    </ul>
  </div>
  <footer>
    <div
      className="tags has-addons"
    >
      <span
        className="tag"
      >
        Connectivity
      </span>
      <span
        className="tag is-primary"
      >
        Online!
      </span>
    </div>
  </footer>
</nav>
`;
